### PR TITLE
Random minor Coverity fixes

### PR DESCRIPTION
**Pull request by danielgustafsson**
A few resource cleanups and a dubious strdup that Coverity was being sad panda about. Nothing terribly exciting but well worth to clean up.

### CODE REVIEW

File: src/backend/postmaster/seqserver.c [hunk 1]
```diff
@@ -988,12 +988,16 @@ listenerSetup(void)
...
+		{
+			freeaddrinfo(rp);
 			continue;
+		}
 
 		if (bind(listenerFd, rp->ai_addr, rp->ai_addrlen) == 0)
 			break;              /* Success */
 
 		close(listenerFd);
+		freeaddrinfo(rp);
```
Comment (hlinnaka) [inline src/backend/postmaster/seqserver.c]: Doesn't this introduce a reference-after-free bug? And shouldn't you only call freeaddrinfo once, to free the whole list, not separately on every element?

File: src/bin/pg_dump/cdb/cdb_ddboost_util.c [hunk 2]
```diff
@@ -3095,7 +3095,7 @@ renameFile(struct ddboost_options *dd_options, ddp_conn_desc_t ddp_conn)
 		return -1;
 	}
 
-	*toFile = Safe_strdup(dd_options->to_file);
+	toFile = Safe_strdup(dd_options->to_file);
```
Comment (hlinnaka) [inline src/bin/pg_dump/cdb/cdb_ddboost_util.c]: Ugh, how can this ever have worked? Makes me think that either this function is in fact never used, or there is no test coverage for it.

File: src/backend/postmaster/seqserver.c (hunk 1 above)
Comment (Eulerizeit) [inline src/backend/postmaster/seqserver.c]: Is there a reason this should be any different from upstream assuming that's where is comes from?

Comment (danielgustafsson) [inline src/backend/postmaster/seqserver.c]: Yes, you're right. I misunderstood the documentation for `getaddrinfo()` and read it as if each member of the linked list should be freed. I didn't manage to break it with this code, but thats not saying too much about the tests. Will fix.

File: src/bin/pg_dump/cdb/cdb_ddboost_util.c (hunk 2 above)
Comment (danielgustafsson) [inline src/bin/pg_dump/cdb/cdb_ddboost_util.c]: I think it isn't used (might still be tried in the field, but it crashed) and it clearly can't be tested. That being said, I'm quite keen to move the tree as little as possible now and instead start off the 6.0 cycle with deprecations like this one. Until then, this patch should make it slightly less bad.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks!

**Pull request closed without merging**